apps/www/registry/default/ui/date-range-picker.tsx

```diff
@@ -0,0 +1,210 @@
+"use client"
+
+import * as React from "react"
+import { Calendar as CalendarIcon } from "lucide-react"
+import { addDays, format } from "date-fns"
+import { DateRange } from "react-day-picker"
+
+import { cn } from "@/lib/utils"
+import { Button } from "@/registry/default/ui/button"
+import { Calendar } from "@/registry/default/ui/calendar"
+import {
+  Popover,
+  PopoverContent,
+  PopoverTrigger,
+} from "@/registry/default/ui/popover"
+import {
+  Select,
+  SelectContent,
+  SelectItem,
+  SelectTrigger,
+  SelectValue,
+} from "@/registry/default/ui/select"
+
+interface DateRangePickerProps extends React.HTMLAttributes<HTMLDivElement> {
+  showExternalPresets?: boolean
+  showInternalPresets?: boolean
+  numberOfMonths?: 2 | 3 | 4 | 5 | 6
+}
+
+export function DateRangePicker({
+  showExternalPresets,
+  showInternalPresets,
+  numberOfMonths = 2,
+  className,
+}: DateRangePickerProps) {
+  const [date, setDate] = React.useState<DateRange | undefined>({
+    from: new Date(2023, 0, 20),
+    to: addDays(new Date(2023, 0, 20), 20),
+  })
+
+  const handlePresetSelect = (value: string) => {

```

Comment:
**Opinion**

Why not leave the actual preset definitions up to the Dev, but rather define a preset interface/type that is then added as an optional array prop to DateRangePickerProps?

Say something like:
```ts
interface DateRangePreset {
  key: string;
  label: string;
  range: {
    from: Date;
    to: Date;
  } 
}

interface DateRangePickerProps extends React.HTMLAttributes<HTMLDivElement> {
  showExternalPresets?: boolean
  showInternalPresets?: boolean
  numberOfMonths?: 2 | 3 | 4 | 5 | 6
  presets?: DateRangePreset[];
}
```

Obviously, you'd loop through the available presets to render appropriately in the `DateRangePicker` component. I'd say that's a better approach than hardcoding preset options (even though that's most likely what a lot of people would go for).

Reply:
It's nice to have presets. Added in the latest commit